platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/GeneratePluginDescriptors.java

```diff
@@ -54,7 +54,12 @@ public GeneratePluginDescriptors() {
         declarations = objectFactory.listProperty(PluginDeclaration.class);
         outputDirectory = objectFactory.directoryProperty();
         implementationClassById = getDeclarations().map(declarations -> declarations.stream()
-            .collect(Collectors.toMap(PluginDeclaration::getId, PluginDeclaration::getImplementationClass, (a, b) -> b, LinkedHashMap::new))
+            .collect(Collectors.toMap(
+                plugin -> plugin.getId().get(),
+                plugin -> plugin.getImplementationClass().get(),
+                (a, b) -> b,
+                LinkedHashMap::new
+            ))
```

Comment:
🤔 You could also keep the original code and prepend with `.map(Provider::get)`. I wonder if that would result in easier to read code. Not really important for this usage alone, but maybe interesting as a general pattern (i.e. "resolve first where we can").

Reply:
Yes, I can do that

Reply:
Hmm, actually this won't work, since `getId()` and `getImplementationClass() were changed to Property 🤔